elementary/files
# Improve tab handling

Fixes #153 

* Separate some Window functions into ViewTab class
* Simplify some related ViewContainer functions
* Less ambiguous Window variable names
* Ensure ViewContainer drops all references when closed.
* Ensure ViewContainer is closed when tab is closed.
* Move some functions to FileUtils
* Improve debugging clarity for destructors

TO TEST:
* Run from the commandline with the -d flag
* Open a single tab _not_ at Home.
* Close this tab.
* The tab is replaced by a "Home" tab and amongst the terminal output are several "--- DESTRUCT ---" messages indicating that the main objects contained in the close tab have been destroyed.

* Open a single tab _not_ at Home.
* Duplicate the tab
* Close the duplicate
* --- DESTRUCT --- messages as above are shown in the terminal.

* Check for regressions related to closing and opening of tabs and windows.
